generate webhook cert without cert-manager (#490)
piraeusdatastore/piraeus-operator: docs/explanation/components.md

# Understanding Piraeus Datastore Components

> Piraeus Datastore consists of seven different components. Every component is running as a separate Deployment, DaemonSet or plain Pod.

Reviewer: ```suggestion
Piraeus Datastore consists of several different components. Each component runs as a separate Deployment, DaemonSet or plain Pod.
```

A number like "seven" is unlikely to be updated when the list is changed.

* Creating storage pools
* Maintaining node labels

# `piraeus-operator-gencert`

The "generate certificate" Pod creates and maintains the TLS key and certificate used by the Piraeus Operator.
The TLS secret, named `webhook-server-cert`, is needed to start the Piraeus Operator Pod. In addition, this Pod
keeps the `ValidatingWebhookConfiguration` for the Piraeus Operator up-to-date.

The Piraeus Operator needs a TLS certificate to serve the validation endpoint for the custom resources it maintains.

Historically, TLS certificates where created by `cert-manager`. This was removed to reduce the number of dependencies
for deploying Piraeus Datastore.

# `linstor-controller`

The LINSTOR Controller is responsible for resource placement, resource configuration, and orchestration of any
operational processes that require a view of the whole cluster. It maintains a database of all the configuration
information for the whole cluster, stored as Kubernetes objects.

The LINSTOR Controller connects to the LINSTOR Satellites and sends them instructions for achieving the desired cluster
state.

It provides an external API used by LINSTOR CSI and Piraeus Operator to change the cluster state.

> # `linstor-satellite`

Reviewer: Perhaps mention why the satellites are created as plain pods? It is unusual.

Author: Done

The LINSTOR Satellite service runs on each node. It acts as the local configuration agent for LINSTOR managed storage.
It is stateless and receives all the information it needs from the LINSTOR Controller.

# `linstor-csi-controller`

> The LINSTOR CSI Controller Pod creates, modifies, deletes volumes and snapshots. It translates the state of Kubernetes

Reviewer: ```suggestion
The LINSTOR CSI Controller Pod creates, modifies and deletes volumes and snapshots. It translates the state of Kubernetes
```
